packages/backend/src/tokens/verify.ts

```diff
@@ -52,3 +64,117 @@ export async function verifyToken(
+        break;
+      case 404:
+        code = MachineTokenVerificationErrorCode.TokenInvalid;
+        message = notFoundMessage;
+        break;
+      default:
+        code = MachineTokenVerificationErrorCode.UnexpectedError;
+        message = 'Unexpected error';
+    }
+
+    return {
+      data: undefined,
+      tokenType,
+      errors: [
+        new MachineTokenVerificationError({
+          message,
+          code,
+          status: err.status,
+        }),
+      ],
+    };
+  }
+
+  return {
+    data: undefined,
+    tokenType,
+    errors: [
+      new MachineTokenVerificationError({
+        message: 'Unexpected error',
+        code: MachineTokenVerificationErrorCode.UnexpectedError,
+        status: err.status,
+      }),
+    ],
+  };
+}
+
+async function verifyMachineToken(
+  secret: string,
+  options: VerifyTokenOptions,
+): Promise<MachineTokenReturnType<MachineToken, MachineTokenVerificationError>> {
+  try {
+    const client = createBackendApiClient(options);
+    const verifiedToken = await client.machineTokens.verifySecret(secret);
+    return { data: verifiedToken, tokenType: TokenType.MachineToken, errors: undefined };
+  } catch (err: any) {
+    return handleClerkAPIError('machine_token', err, 'Machine token not found');
+  }
+}
+
+async function verifyOAuthToken(
+  secret: string,
+  options: VerifyTokenOptions,
+): Promise<MachineTokenReturnType<IdPOAuthAccessToken, MachineTokenVerificationError>> {
+  try {
+    const client = createBackendApiClient(options);
+    const verifiedToken = await client.idPOAuthAccessToken.verifySecret(secret);
+    return { data: verifiedToken, tokenType: TokenType.OAuthToken, errors: undefined };
+  } catch (err: any) {
+    return handleClerkAPIError(TokenType.OAuthToken, err, 'OAuth token not found');
+  }
+}
+
+async function verifyAPIKey(
+  secret: string,
+  options: VerifyTokenOptions,
+): Promise<MachineTokenReturnType<APIKey, MachineTokenVerificationError>> {
+  try {
+    const client = createBackendApiClient(options);
+    const verifiedToken = await client.apiKeys.verifySecret(secret);
+    return { data: verifiedToken, tokenType: TokenType.ApiKey, errors: undefined };
+  } catch (err: any) {
+    return handleClerkAPIError(TokenType.ApiKey, err, 'API key not found');
+  }
+}
+
+/**
+ * Verifies any type of machine token by detecting its type from the prefix.
+ *
+ * @param token - The token to verify (e.g. starts with "m2m_", "oauth_", "api_key_", etc.)
+ * @param options - Options including secretKey for BAPI authorization
+ */
+export async function verifyMachineAuthToken(token: string, options: VerifyTokenOptions) {
+  if (token.startsWith(M2M_TOKEN_PREFIX)) {
+    return verifyMachineToken(token, options);
+  }
+  if (token.startsWith(OAUTH_TOKEN_PREFIX)) {
+    return verifyOAuthToken(token, options);
+  }
+  if (token.startsWith(API_KEY_PREFIX)) {
+    return verifyAPIKey(token, options);
+  }
+
+  throw new Error('Unknown machine token type');
+}
```

Comment:
This is the main function that verifies the incoming machine token by checking its prefix  and routing to the appropriate verification function.